string startNode
string endNode
---
int32 len